imports/plugins/core/orders/client/templates/workflow/shippingInvoice.js

```diff
@@ -58,14 +59,33 @@ Template.coreOrderShippingInvoice.events({
 
     const state = instance.state;
     const order = state.get("order");
+    const orderTotal =
+      order.billing[0].invoice.subtotal
+      + order.billing[0].invoice.shipping
+      + order.billing[0].invoice.taxes;
     const discount = state.get("field-discount") || 0;
 
-    Meteor.call("orders/approvePayment", order, discount, (error) => {
-      if (error) {
-        // Show error
-        Logger.warn(error);
-      }
-    });
+    if (discount > orderTotal) {
+      Alerts.inline("Discount cannot be greater than original total price", "error", {
+        placement: "coreOrderShippingInvoice",
+        i18nKey: "order.invalidDiscount",
+        autoHide: 10000
+      });
+    } else {
+      Meteor.call("orders/approvePayment", order, discount, (error) => {
+        if (error) {
+          // Log & show error if client-side fails
```

Comment:
Are you not calling a server-side method and that is in fact what would fail there?

No we don't want to pass on any secure information that may leak out of an unredacted error message. We could however check for "expected" (if any) errors and pass on friendly messages.